src/services/content/hooks/materials.js

```diff
@@ -0,0 +1,112 @@
+const { authenticate } = require('@feathersjs/authentication');
+const { Forbidden } = require('@feathersjs/errors');
+const { hasPermission } = require('../../../hooks');
+const { getScopePermissions } = require('../../helpers/scopePermissions/hooks/checkScopePermissions');
+
+/**
+ * Returns true if the user making the request has the required permissions
+ * in all courses which reference the given material.
+ * @param {Context} context hook context
+ * @param {ObjectId} id Id of the material
+ * @param {Array<String>} permissions Array of required permissions
+ * @returns {Boolean} true if conditions are met, false otherwise
+ */
+const hasMaterialAccess = async (context, id, permissions) => {
+	const lessons = await context.app.service('lessons').find({
+		query: {
+			materialIds: id,
+			$select: ['courseId'],
+		},
+		paginate: false,
+	});
+	const courses = [...new Set(lessons.map((l) => l.courseId))];
+
+	let access = true;
+	for (const courseId of courses) {
+		const scope = { id: courseId, name: 'courses' };
+		let userPermissions;
+		try {
+			userPermissions = await getScopePermissions(context.app, context.params.account.userId, scope);
+		} catch (err) {
+			// the course scope throws Forbidden if the user is not in the course (?!)
+			userPermissions = [];
+		}
+		access = access && permissions.every((permission) => userPermissions.includes(permission));
+		if (!access) break;
+	}
+	return access;
+};
+
+/**
+ * Before-hook to enforce the requesting user has the required permisions in the course
+ * the material is used in. If the material is used in multiple courses, the permissions
+ * will be required in all of them.
+ * @requires authenticate('jwt')
+ * @param  {...String} permissions list of permissions necessary for this hook to resolve
+ * @returns {Context} hook context
+ */
+const checkAssociatedCoursePermission = (...permissions) => async (context) => {
+	if (permissions) {
```

Comment:
Make this hook sense without permissions? I think we can remove the if permissions, that should fail if no permissions is passed.